main.nf

```diff
@@ -1131,112 +1131,119 @@ process busco_download_db {
+    busco_path=\$(which busco)
+    config_file="\${busco_path%bin/busco}config/config.ini"
+
+    if [ ${cp_augustus_config} = "Y" ] ; then
         cp -r /opt/conda/pkgs/augustus*/config augustus_config/
         export AUGUSTUS_CONFIG_PATH=augustus_config
+    fi
 
-        run_BUSCO.py \
-            --in ${assembly} \
-            --lineage_path $db_name \
-            --cpu "${task.cpus}" \
-            --blast_single_core \
-            --mode genome \
-            --out ${assembly} \
-            >${assembly}_busco_log.txt
-        cp run_${assembly}/short_summary_${assembly}.txt short_summary_${assembly}.txt
-
-        for f in run_${assembly}/single_copy_busco_sequences/*faa; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*faa >${assembly}_buscos.faa || touch ${assembly}_buscos.faa
-            break
-        done
-        for f in run_${assembly}/single_copy_busco_sequences/*fna; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*fna >${assembly}_buscos.fna || touch ${assembly}_buscos.fna
-            break
-        done
-        """
-    } else {
-        """
-        run_BUSCO.py \
-            --in ${assembly} \
-            --lineage_path $db_name \
-            --cpu "${task.cpus}" \
-            --blast_single_core \
-            --mode genome \
-            --out ${assembly} \
-            >${assembly}_busco_log.txt
-        cp run_${assembly}/short_summary_${assembly}.txt short_summary_${assembly}.txt
-
-        for f in run_${assembly}/single_copy_busco_sequences/*faa; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*faa >${assembly}_buscos.faa || touch ${assembly}_buscos.faa
-            break
-        done
-        for f in run_${assembly}/single_copy_busco_sequences/*fna; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*fna >${assembly}_buscos.fna || touch ${assembly}_buscos.fna
-            break
-        done
-        """
-    }
+    # place db in extra folder to ensure BUSCO recognizes it as path (instead of downloading it)
+    mkdir dataset
+    mv ${db} dataset/
+
+    busco \
+        --mode genome \
+        --in ${bin} \
+        --lineage_dataset dataset/${db} \
+        --config \${config_file} \
+        --cpu "${task.cpus}" \
+        --out "BUSCO" > ${bin}_busco.log
+
+    cp BUSCO/short_summary.specific.${db}.BUSCO.txt short_summary.specific.${db}.${bin}.txt
+
+    for f in BUSCO/run_${db}/busco_sequences/single_copy_busco_sequences/*faa; do
+        [ -e "\$f" ] && cat BUSCO/run_${db}/busco_sequences/single_copy_busco_sequences/*faa >${bin}_buscos.faa
+        break
+    done
+    for f in BUSCO/run_${db}/busco_sequences/single_copy_busco_sequences/*fna; do
+        [ -e "\$f" ] && cat BUSCO/run_${db}/busco_sequences/single_copy_busco_sequences/*fna >${bin}_buscos.fna
+        break
+    done
```

Comment:
You removed the failsave `|| touch ${assembly}_buscos.fna`, does that mean its always there? In this case `[ -e "\$f" ] && ` might be not needed as well? I am unsure here.

Reply:
I added `optional true` to the output, thought it might be nicer than having empty files in the results folder. 
`[ -e "\$f" ]`  is still needed, as it could be that the file does not exist, and than `cat` should not be applied

Reply:
Alright, sounds good.